### ISSUE TITLE
Static linking (Windows)

### ISSUE THREAD

**Issue by John4266** [whoshuu/cpr#478]
Hello.
I am trying to use a static version of the library installed through VCPKG using the command`./vcpkg install cpr:x86-windows-static`
but i am getting many errors.
![image](https://user-images.githubusercontent.com/70203256/99446741-b0b44680-2927-11eb-8768-363264c6d2cc.png)

This are my settings:
![image](https://user-images.githubusercontent.com/70203256/99446293-7d71b780-2927-11eb-985f-6ec84ff1a3b9.png)
![image](https://user-images.githubusercontent.com/70203256/99446327-89f61000-2927-11eb-94f4-79d2f448c591.png)

**Comment by COM8** [comment]
Thanks for reporting. I think this is a problem for non x64 systems on windows.

**Comment by John4266** [comment]
> 
> 
> Thanks for reporting. I think this is a problem for non x64 systems on windows.

nah, it also happens when using x64

**Comment by COM8** [comment]
Arg, ok.

**Comment by John4266** [comment]
uh, any eta for fix?

**Comment by COM8** [comment]
Not really since I do not know how to handle VS on Windows in CMAKE.
As a workarround I suggest using vcpkg and the version of CPR from there. This should work.

@KingKili do you have any idea how to fix this?

**Comment by KingKili** [comment]
I think vcpg is already in use. The cpr buildsystem should respect the variable 'BUILD_SHARED_LIBS', therefore I would try to use to set it to false for the configuration run of cpr.

**Comment by John4266** [comment]
yea, i am already using cpr from vcpkg.

**Comment by Kid0h** [comment]
Did you find a solution by any chance?
Currently trying to myself, but I don't really expertise in linking and stuff like that..

**Comment by John4266** [comment]
> 
> 
> Did you find a solution by any chance?
> Currently trying to myself, but I don't really know how to work with linking and stuff like that..

Unfortunately, no.
I think its related to the Curl setup.

**Comment by Kid0h** [comment]
Maybe any workarounds? 
Are you able to do the linking with the source code itself added as a library? (Instead of using vcpkg..)

**Comment by John4266** [comment]
> 
> 
> Maybe any workarounds?
> Are you able to do the linking with the source code itself added as a library? (Instead of using vcpkg..)

I tried it as well a while ago.
btw אני גם ישראלי

**Comment by Kid0h** [comment]
> > Maybe any workarounds?
> > Are you able to do the linking with the source code itself added as a library? (Instead of using vcpkg..)
> 
> I tried it as well a while ago with no luck (Maybe i was doing it wrong).
> btw אני גם ישראלי

What's your Discord @?
I might have a workaround I can that I'm currently testing.

**Comment by Kid0h** [comment]
@COM8 Is this being investigated?
This has been an issue for months.

**Comment by COM8** [comment]
TLDR: No, not from our side.
We do not use Windows and have never been using VS with an CMake project/static libs in C++.

Here we probably need help from people who actually use such a system. You are also more than welcome to investigate this your self and create a PR then.

I guess here we need some kind of CMake magic to fix the VS/Windows linking stuff.

**Comment by MikeGachanja** [comment]
Has this issue been solved yet? It has been running for maybe too long?

**Comment by MikeGachanja** [comment]
Has this issue been solved yet? It has been running for maybe too long?

**Comment by COM8** [comment]
Sadly no. Looks like nobody knows how Windows or especially VS linking works with CMake.

**Comment by COM8** [comment]
Closing this one in favor of #361

**Comment by GsoyG** [comment]
I encountered a similar problem, but I found a solution in curl's issues.

The reason is probably because curl under the Windows platform calls the Win32 API. If you statically link curl, you must also statically link the library where the Win32 API is located. This does not happen automatically, so the Win32 API symbols cannot be found.

**Solution: In the project properties of VS, Linker-Input-Additional Dependencies. Add the following four static library names:**
```
WS2_32.Lib
Wldap32.Lib
Crypt32.Lib
Normaliz.lib
```
